apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/common/ClaimDao.java

```diff
@@ -124,6 +136,52 @@ Predicate createDateRangePredicate(
     return QueryUtils.createLastUpdatedPredicateInstant(builder, root, dateRange);
   }
 
+  @VisibleForTesting
+  Predicate serviceDateRangePredicate(
+      Root<?> root,
+      DateRangeParam serviceDate,
+      CriteriaBuilder builder,
+      String endDateAttributeName) {
+    Path<LocalDate> serviceDateEndPath = root.get(endDateAttributeName);
+
+    List<Predicate> predicates = new ArrayList<>();
+
+    // If the row's end service date attribute is null, don't return it
+    predicates.add(builder.isNotNull(serviceDateEndPath));
+
```

Comment:
Interesting.  Wouldn't a NULL be normally be omitted in a query result if there is any condition on that column?

Reply:
Maybe this was a protection in case the user DIDNT specify a `service-date` filter?  In that case I think we should remove this predicate, just in case we want to integrate with other peering APIs down the road that DONT use the `service-date` query param

Reply:
This line checks to see if the service end date in the database is null